internal/helmdeployer/helmcache/secret.go

```diff
@@ -3,25 +3,25 @@ package helmcache
 import (
 	"context"
 
-	corecontrollers "github.com/rancher/wrangler/v2/pkg/generated/controllers/core/v1"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/watch"
 	applycorev1 "k8s.io/client-go/applyconfigurations/core/v1"
 	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 // SecretClient implements methods to handle secrets. Get and list will be retrieved from the wrangler cache, the other calls
 // will be make to the Kubernetes API server.
 type SecretClient struct {
-	cache     corecontrollers.SecretCache
+	cache     client.Client
```

Comment:
Well, as opposed to the client, this is the cached client.

Reply:
The `client.Client` is provided by controller-runtime. The package is still a caching storage driver for helm, even if we inject dependencies. If you have a better name, please, but I think `helmcache` is to the point and short enough. This PR didn't change anything about that, the `corecontrollers.SecretCache` did all the caching before and it's also injected.

Reply:
But in that case, the type was `SecretCache` and not `SecretClient`.
I've tried to follow where this `client.Client` is coming from (its part of the `Helm` object, which takes it from the `localClient`, received from this code https://github.com/rancher/fleet/blob/a138a30d1268d0cdb30343f54ed170bd0dbe534d/internal/cmd/agent/controller.go#L228), and there is no explicit mention to this being a cached content. It completely relies on the default settings, which could inadvertently change.
I'm not against changing the method name, but to ensure that the client used by `SecretClient` really caches secrets.

Reply:
This is the main client we're going to use from now on. Construction is a bit special, because we need to build it for upstream by ourselves. Otherwise the controller-runtime manager does that for us.

> It is a common pattern in Kubernetes to read from a cache and write to the API server. This pattern is covered by the creating the Client with a Cache.  (https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client
)

You already linked `newCluster` where I [set up the cache](https://github.com/rancher/fleet/blob/a138a30d1268d0cdb30343f54ed170bd0dbe534d/internal/cmd/agent/controller.go#L236-L242), similar to the code in the manager.


Again, this isn't much different than before, instead of `client.Client`, we relied on `"github.com/rancher/wrangler/v2/pkg/generated/controllers/core/v1".SecretCache` to stay cached.